test/rtos_drivers/usb/src/FreeRTOSIPConfig.h

```diff
@@ -1,372 +1,374 @@
-responsibility of the Ethernet interface to filter out packets that are of no
-interest.  If the Ethernet interface does not implement this functionality, then
-set ipconfigETHERNET_DRIVER_FILTERS_FRAME_TYPES to 0 to have the IP stack
-perform the filtering instead (it is much less efficient for the stack to do it
-because the packet will already have been passed into the stack).  If the
-Ethernet driver does all the necessary filtering in hardware then software
-filtering can be removed by using a value other than 1 or 0. */
-#define ipconfigETHERNET_DRIVER_FILTERS_FRAME_TYPES	1
-
-/* The windows simulator cannot really simulate MAC interrupts, and needs to
-block occasionally to allow other tasks to run. */
-#define configWINDOWS_MAC_INTERRUPT_SIMULATOR_DELAY ( 20 / portTICK_PERIOD_MS )
-
-
-#ifndef SL_WFX_USE_SECURE_LINK
-/* Allocate space before Ethernet frames to hold the WF200 send frame
- * request header. This allows FreeRTOS+TCP network buffers to be sent
- * directly to the WF200 driver without allocating a new buffer and copying
- * the frame from the FreeRTOS+TCP network buffer into it.
- *
- * The + 2 is so that the 14 byte Ethernet frame starts 2 bytes after a 32-bit
- * boundary so that the IP header can begin on one. */
-#define ipconfigPACKET_FILLER_SIZE ( sizeof(sl_wfx_send_frame_req_t) + 2 )
-#endif
-
-/* Define the size of the pool of TCP window descriptors.  On the average, each
-TCP socket will use up to 2 x 6 descriptors, meaning that it can have 2 x 6
-outstanding packets (for Rx and Tx).  When using up to 10 TP sockets
-simultaneously, one could define TCP_WIN_SEG_COUNT as 120. */
-#define ipconfigTCP_WIN_SEG_COUNT		20
-
-/* Each TCP socket has a circular buffers for Rx and Tx, which have a fixed
-maximum size.  Define the size of Rx buffer for TCP sockets. */
-#define ipconfigTCP_RX_BUFFER_LENGTH			( 4 * ipconfigTCP_MSS )
-
-/* Define the size of Tx buffer for TCP sockets. */
-#define ipconfigTCP_TX_BUFFER_LENGTH			( 4 * ipconfigTCP_MSS )
-
-/* Limit the amount of TCP segments sent out at a time per socket.
-The default of 8 makes it very difficult for more than one output
-stream to operate at a time. Setting it to 1 allows them to better
-share time. */
-#define SEND_REPEATED_COUNT     ( 8 )
-
-/* When using call-back handlers, the driver may check if the handler points to
-real program memory (RAM or flash) or just has a random non-zero value. */
-#define ipconfigIS_VALID_PROG_ADDRESS(x) ( (x) != NULL )
-
-/* Include support for TCP hang protection.  All sockets in a connecting or
-disconnecting stage will timeout after a period of non-activity. */
-#define ipconfigTCP_HANG_PROTECTION			( 1 )
-#define ipconfigTCP_HANG_PROTECTION_TIME	( 30 )
-
-/* Include support for TCP keep-alive messages. */
-#define ipconfigTCP_KEEP_ALIVE				( 1 )
-#define ipconfigTCP_KEEP_ALIVE_INTERVAL		( 20 ) /* in seconds */
-
-#define portINLINE __inline
-
-#endif /* FREERTOS_IP_CONFIG_H */
+// Copyright 2019-2023 XMOS LIMITED.
```

Comment:
I have concerns about adding an XMOS copyright and license statement when Amazon have already claimed copyright and provided a license.

Reply:
This file is part of a test, just to be precise.

Reply:
Having read the Amazon license, it includes the right to sub-license and to use without restriction.  I hadn't taken the time to notice that before.  I agree that the XMOS copyright and licence notices can remain.